### DISCUSSION TITLE
Ziggy route filtering for Inertia app(react, laravel)

### DISCUSSION THREAD

**Discussion by janyalda17** [tighten/ziggy#647] [Q&A]
I've been attempting to filter out some routes that should only be visible to authenticated users, following the [Ziggy docs](https://github.com/tighten/ziggy#filtering-routes) I'm able to group the routes and use the `@routes` directive passing the group name in `app.blade.php` to only share those routes.

But where I'm having trouble is after logging in they don't change and I get Ziggy front-end errors until I refresh the page and I get the right routes then it works, this is what I initially thought would do it which is the following conditions in `app.blade.php` :
``` php
@auth
    @routes('authenticated')
@endauth
@guest
    @routes('guests')
@endguest
```

### So what I'm wondering is if there is a way to dynamically send back the needed routes based on the auth check at the point where a user is logging in, or is this not possible?

**Comment by bakerkretzmar** [comment]
Yes, that's possible. How exactly you do it is going to depend on your app setup and build/framework, but there are several examples in other discussions: #465, https://github.com/tighten/ziggy/discussions/463#discussioncomment-1381011.

**Reply by janyalda17** [reply]
Yea, I did run into those examples while looking into this matter and it does work while passing it back with the shared data but I was having trouble updating the routes on the front-end to match whats passed back and I halted that approach to attempt something else but now that you mentioned it again I'll give it another try. 

Thanks Jacob! 👍
